CUDADataFormats/Common/interface/CUDA.h

```diff
@@ -0,0 +1,81 @@
+#ifndef CUDADataFormats_Common_CUDA_h
+#define CUDADataFormats_Common_CUDA_h
+
+#include <optional>
+
+#include <cuda/api_wrappers.h>
+
+namespace edm {
+  template <typename T> class Wrapper;
+}
+
+/**
+ * The purpose of this class is to wrap CUDA data to edm::Event in a
+ * way which forces correct use of various utilities.
+ *
+ * The non-default construction has to be done with CUDAScopedContext
+ * (in order to properly register the CUDA event).
+ *
+ * The default constructor is needed only for the ROOT dictionary generation.
+ *
+ * The CUDA event is in practice needed only for stream-stream
+ * synchronization, but someone with long-enough lifetime has to own
+ * it. Here is a somewhat natural place. If overhead is too much, we
+ * can e.g. make CUDAService own them (creating them on demand) and
+ * use them only where synchronization between streams is needed.
+ */
+template <typename T>
+class CUDA {
```

Comment:
I agree, the name could be better (but it should still be relatively short, `CUDAWrapper`?). The intended use is
```
 * The purpose of this class is to wrap CUDA data to edm::Event in a
 * way which forces correct use of various utilities.
```
For more information see
* slide 10 of https://indico.cern.ch/event/746161/contributions/3084531/attachments/1692036/2722511/slides_mk_20180724.pdf
* https://github.com/cms-patatrack/cmssw/blob/0800aac1eeea41ccb73826a4315c82a0f1a6f8cd/HeterogeneousCore/CUDACore/README.md#data-model

Reply:
Or what about `edm::CUDA`? (or is that still too generic name?)

Reply:
I would say `edm::CUDA` is sill too generic, since it is not an interface to the whole CUDA framework.
I would be OK with `CUDAWrapper` or `CUDADataWrapper`.

Reply:
How about `CUDAProduct`? It would be a bit more specific than `...Wrapper` (and could not be confused with `edm::Wrapper`).